bpf-gpl/parsing.h

```diff
@@ -19,7 +19,20 @@
 #define __CALI_PARSING_H__
 
 static CALI_BPF_INLINE int parse_packet_ip(struct cali_tc_ctx *ctx) {
-	switch (bpf_htons(ctx->skb->protocol)) {
+	__u16 protocol = 0;
+
+	if (CALI_F_XDP) {
+		if (skb_refresh_validate_ptrs(ctx, UDP_SIZE)) {
```

Comment:
It feels odd to need an skb_refresh_validate_ptrs call in the XDP case but not for TC.  Could the refresh call be moved up into the caller, so that we can have the same assumption - i.e. that ctx pointers are valid - for both XDP and TC here?

Reply:
Oh I see, you have addressed this below - please ignore this comment.